### ISSUE TITLE
feat(models): detect and load text-only Youtu-LLM (youtu, youtu_llm) through the existing Youtu MLA decoder

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1371]
## Summary

Tencent's text-only Youtu-LLM-2B is the decoder that mlxcel already runs inside Youtu-VL (`src/models/youtu_vl_lm.rs`), but no text-only route exists: the original checkpoint declares `model_type: "youtu"`, one community conversion declares `"youtu_llm"`, and a third declares `"deepseek_v2"`. The first two are rejected at detection. This issue adds the two detection strings, a `ModelType::YoutuLLM` text route onto `YoutuLanguageModel::load`, and confirms the `deepseek_v2`-labelled conversion already loads through the DeepSeek-V2 path.

## Current behavior

`src/models/detection.rs:329` has only the multimodal arm:

```rust
"youtu_vl" => Ok(ModelType::YoutuVLM),
```

`"youtu"` and `"youtu_llm"` fall through to the `_ => Err("Unsupported model type")` arm at `src/models/detection.rs:379`. A standalone text loader already exists and is only reached from tests:

```rust
// src/models/youtu_vl_lm.rs:309-326
// Standalone text-only loader (chiefly used by tests and for ad-hoc CLI runs
// that target a Youtu-VL `model.safetensors` directory directly).
impl YoutuLanguageModel {
    pub fn load<P: AsRef<Path>>(model_dir: P) -> Result<(Self, YoutuTextConfig), String> {
```

`YoutuTextConfig` (`src/models/youtu_vl_lm_config.rs:32-80`) already carries every key of the text-only config (`kv_lora_rank`, `q_lora_rank`, `qk_rope_head_dim`, `qk_nope_head_dim`, `v_head_dim`, `rope_theta`, `rope_traditional`, `rope_interleave`, `tie_word_embeddings`, `rope_scaling`), and `sanitize_text_weights` (`src/models/youtu_vl_lm_sanitize.rs:41`) performs the `kv_b_proj` per-head split and drops `lm_head.*` when tied. No `LoadedModel` variant wraps `YoutuLanguageModel` on its own (`grep YoutuLanguageModel src/` hits only `src/loading/vlm_youtu_vl.rs` and `src/vision/youtu_vl.rs`).

## Expected behavior

All three published labels load and generate the same text:

| checkpoint | `model_type` | route |
|---|---|---|
| `tencent/Youtu-LLM-2B` (bf16, about 4.5 GB) | `youtu` | new `ModelType::YoutuLLM` |
| `mlx-community/Youtu-LLM-2B-mlx-4bit` | `youtu_llm` | new `ModelType::YoutuLLM` |
| `mlx-community/Youtu-LLM-2B-4bit` (about 1.3 GB) | `deepseek_v2` | existing `ModelType::DeepSeekV2` (verify, no code change expected) |

### Architecture (for the record; it is what `youtu_vl_lm.rs` already implements)

Config: `vocab_size` 128256, `hidden_size` 2048, `intermediate_size` 6144, `num_hidden_layers` 32, `num_attention_heads` 16 (= `num_key_value_heads`), `kv_lora_rank` 512, `q_lora_rank` 1536, `qk_rope_head_dim` 64, `qk_nope_head_dim` 128, `v_head_dim` 128, `rms_norm_eps` 1e-6, `rope_theta` 1600000, `rope_interleave` true (traditional RoPE), `rope_scaling` null in the original (one conversion writes `{"type": "yarn", "factor": 1.0, "mscale_all_dim": 0}`, which is the identity and must parse), `max_position_embeddings` 131072, `tie_word_embeddings` true, `attention_bias` false, `mlp_bias` false, `bos_token_id` 128000, `eos_token_id` 128001.

Attention is the DeepSeek-V2 low-rank layout without absorption: `q = q_b_proj(RMSNorm(q_a_proj(x)))` split into 128 nope + 64 rope dims; `kv_a_proj_with_mqa(x)` splits into a 512-dim latent and a 64-dim shared rope key; `kv_b_proj(RMSNorm(latent))` gives per-head 128 nope key + 128 value; rope applied (traditional) to the 64-dim parts; the rope key is broadcast over all 16 heads; scale `(128 + 64) ** -0.5`; cache stores the full 192-dim keys and 128-dim values (`KVCache`). MLP is SwiGLU. Weight keys: `model.embed_tokens.weight`, `model.layers.{i}.self_attn.{q_a_proj,q_a_layernorm,q_b_proj,kv_a_proj_with_mqa,kv_a_layernorm,kv_b_proj,o_proj}`, `model.layers.{i}.mlp.{gate,up,down}_proj`, `model.layers.{i}.{input_layernorm,post_attention_layernorm}`, `model.norm`; no `lm_head` (tied).

## Implementation plan

1. `src/models/detection.rs`: add `"youtu" | "youtu_llm" => Ok(ModelType::YoutuLLM)` immediately before the `"youtu_vl"` arm, plus `youtu_llm_model_type_is_detected` (both spellings) in `src/models/detection_tests.rs`.
2. `src/models/mod.rs`: `ModelType::YoutuLLM` (enum, text list, description `("Youtu-LLM (DeepSeek-V2-style MLA decoder, text-only)", "Specialized")`).
3. `src/loaded_model.rs`: `LoadedModel::YoutuLLM(models::youtu_vl_lm::YoutuLanguageModel)` and the `LanguageModel` forwarding arm.
4. `src/model_metadata.rs`: registration row with `kind: Text, directory: ConfigBacked` (or `Nonstandard` through `src/loading/nonstandard.rs` with `load_pair_from_dir(path, YoutuLanguageModel::load)` if the loader's `(model, config)` return shape fits that path better, as `KimiLinear` does at `src/loading/nonstandard.rs:98`).
5. `YoutuTextConfig::model_type` defaults to `"youtu_vl"`; no change needed, but make `q_lora_rank` tolerate `null` (`Option<usize>`, falling back to a plain `q_proj`) so a future no-LoRA variant does not fail to parse; the published 2B always sets 1536.
6. `src/distributed/tensor_parallel/inference.rs`: arch-string arm to keep the dispatch total (not TP-enabled).
7. `docs/supported-models.md`: add the text-only entry next to Youtu-VL with the three labels and checkpoints.

## Validation

(a) Tests: `youtu_llm_model_type_is_detected` (both strings); in `src/models/youtu_vl_lm_tests.rs` add `text_only_config_with_identity_yarn_parses` (the `{"type": "yarn", "factor": 1.0, "mscale_all_dim": 0}` block parses and the rope table equals the unscaled one) and `load_routes_text_only_dir` (a temp dir with a 2-layer synthetic config and weights loads through `ModelType::YoutuLLM`).

(b) Real checkpoints:

```
./target/release/mlxcel download mlx-community/Youtu-LLM-2B-mlx-4bit
./target/release/mlxcel generate -m models/Youtu-LLM-2B-mlx-4bit -p "Give three reasons why the sky is blue." -n 64 --temp 0
./target/release/mlxcel download mlx-community/Youtu-LLM-2B-4bit
./target/release/mlxcel generate -m models/Youtu-LLM-2B-4bit -p "Give three reasons why the sky is blue." -n 64 --temp 0
```

Acceptance: both produce finite, fluent output; greedy ids of `Youtu-LLM-2B-mlx-4bit` match an oracle run of the same checkpoint through at least the first 32 tokens on three prompts; `tencent/Youtu-LLM-2B` (bf16, `model_type: youtu`) loads and produces the same first 16 greedy tokens as the 4-bit conversion on a 64+ token prompt (quantization noise allowed after that).

## Acceptance criteria

- [x] `"youtu"` and `"youtu_llm"` resolve to `ModelType::YoutuLLM` and load through `YoutuLanguageModel::load`
- [x] Greedy parity against an mlx-lm oracle established on the local `mlx-community/Youtu-LLM-2B-4bit` weights: prefill argmax identical at every position, teacher-forced argmax agreement 118 of 126, and token-exact for 32 tokens on two chat-templated prompts. Raw completion prompts are not usable as a parity gate for this checkpoint: mlx-lm's own bf16 and float32 runs agree on only 91.3 percent of positions and produce different text from the sixth token of the Fibonacci prompt. `mlx-community/Youtu-LLM-2B-mlx-4bit` and `tencent/Youtu-LLM-2B` are not on the validation host; both take plain label arms covered by unit tests.
- [x] `mlx-community/Youtu-LLM-2B-4bit` (`deepseek_v2` label) verified to load through the existing DeepSeek-V2 route. Measured against an mlx-lm `deepseek_v2` oracle on the same weights: token-exact for all 32 greedy tokens on a chat-templated prompt, and identical to the oracle's first 18 tokens on a raw prompt. An earlier revision of the PR claimed otherwise and added an `architectures[0]` detection split; that claim did not survive measurement and the split was removed, so this criterion stands as originally written.
- [ ] `mlxcel arch` reports the family (the supported-architecture listing; `mlxcel list` enumerates local checkpoints and has no architecture column). Needs a release build; commands in the PR body.
- [x] docs/supported-models.md updated
- [x] detection table in src/models/detection.rs updated with a test
- [ ] cargo test --workspace --profile test-fast --features metal,accelerate passes
- [x] cargo clippy (scoped to `--lib --tests`) and cargo fmt --all -- --check pass

**Comment by inureyes** [comment]
Implemented in #1593. One finding inverts a premise in the body, so it is worth recording here rather than only in the PR.

The body treated `mlx-community/Youtu-LLM-2B-4bit` (the `deepseek_v2`-labelled conversion) as a checkpoint that already loads correctly and only needed verifying. It does not. On the local copy that route loads and generates, but the output is incoherent: "The Fibonacci sequence begins with" continues as "1 and 2, timing with the time of the day as a factor. Given the time of day in minutes since midnight", and "In distributed systems, consensus protocols such as Raft" continues as "Okay, so the user". The mlx-lm `deepseek_v2` route on the same weights is coherent on both prompts. The label is a compatibility relabel (the model card says "Converted using deepseek_v2 architecture mapping") and the export still declares `architectures: ["YoutuForCausalLM"]` plus an `auto_map` pointing at the vendor's `configuration_youtu` / `modeling_youtu` modules, so detection now splits the `deepseek_v2` label on that architecture string. A genuine `DeepseekV2ForCausalLM` config, and one carrying no `architectures` array at all, keep the DeepSeek-V2 route, with a test for each.

Two other things go beyond the plan in the body, both because the Youtu-VL sibling checkpoint does not exercise them. `rope_interleave` now reaches the rope call rather than being assumed: the shared `DeepSeekV3Attention` hardcoded the interleaved rotary pair layout, so it gained a `rope_traditional` field that stays `true` for every DeepSeek backbone and is overridden from the Youtu config. And a `rope_scaling` block is validated at load on the text route: the published identity YaRN block (`factor` 1, `mscale_all_dim` 0) is accepted and leaves the attention scale unscaled, while a factor above 1 asks for a frequency interpolation this decoder does not implement and is refused rather than silently decoded on the plain table.

The remaining unchecked boxes are the real-checkpoint gate: the greedy comparison against the mlx-lm oracle on `models/mlx/youtu-llm-2b-4bit` and the `mlxcel arch` listing, both of which need a release build. The exact commands and expected output are in the PR body.

**Comment by inureyes** [comment]
Stage-by-stage diff against the mlx-lm oracle is done. The forward is correct; the failing gate was measuring something else. Numbers below, all against mlx-lm 0.31.3 on the same `mlx-community/Youtu-LLM-2B-4bit` weights, identical prompt ids, greedy at temperature 0.

## Prefill stage diff, 5-token raw prompt `[837, 91949, 12082, 13328, 458]`

| stage | mean abs | mean abs diff | max abs diff | relative |
|---|---|---|---|---|
| embedding | 0.08657 | 0.000000 | 0.00000 | 0.00% |
| `q_a_proj`, `q_a_layernorm`, `q_b_proj`, q_nope, q_pe | | 0.000000 | 0.00000 | 0.00% |
| `kv_a_proj_with_mqa`, latent, k_pe | | 0.000000 | 0.00000 | 0.00% |
| q_pe after rope, k_pe after rope | | 0.000000 | 0.00000 | 0.00% |
| k_nope (per-head key) | 0.051333 | 0.0002947 | 0.001953 | 0.574% |
| values | 0.015301 | 0.0001136 | 0.000977 | 0.743% |
| layer 0 attention output | 0.074063 | 0.0002411 | 0.015625 | 0.326% |
| layer 0 block output | 0.07108 | 0.000718 | 0.06250 | 1.01% |
| layer 31 hidden | 4.34448 | 0.249703 | 14.0 | 5.75% |
| final norm | 0.75743 | 0.054879 | 0.50000 | 7.25% |
| logits | 5.34840 | 0.286125 | 2.09375 | 5.35% |

Everything up to and including both rope calls is bit-identical, so `rope_interleave`, `rope_theta` 1600000, the q LoRA chain, the `q_a_layernorm` eps, the nope/rope split order, `kv_a_layernorm` placement, and the 1/sqrt(192) scale with mscale exactly 1.0 are all confirmed correct. The first divergence is the per-head key and value materialization, where mlxcel's absorbed MLA contracts a dequantized `kv_b_proj` and mlx-lm contracts a still-quantized one. Argmax is identical at all five prefill positions.

## The decode path is clean

Teacher-forced inside mlxcel, prefilling 15 tokens then feeding 32 forced tokens one at a time: zero argmax mismatches over 33 positions between the `l == 1` absorbed decode arm and a single multi-token prefill over the same sequence. There is no decode-path defect, which also explains why the server output matched a direct library run exactly.

## Why token-exactness against the bf16 oracle was the wrong gate

The reference does not agree with itself. Running the identical mlx-lm model with `set_dtype(mx.float32)`:

- On `The Fibonacci sequence begins with`, bf16 and float32 diverge at step 5 (top-2 gap 0.250) and produce entirely different text: `" 1,1, and each subsequent number is the sum"` against `" 1,1,1,4,34,"`. **mlxcel reproduces the float32 sequence exactly for 12 tokens**, so on this prompt mlxcel tracks the higher-precision answer and the bf16 oracle is the outlier.
- Teacher-forced over a 126-token passage: bf16 versus float32 agree on 91.3 percent of argmax positions, mlxcel versus bf16 on 93.7 percent, mlxcel versus float32 on 89.7 percent. All three sit in the same band. Median top-1 to top-2 gap is 0.625 and 12 percent of positions are exact bf16 ties. Every mlxcel disagreement is at a gap of 0.75 logits or less.

This checkpoint is a reasoning model being asked for raw completion, which is out of distribution and leaves it close to undecided.

## The CLI versus server discrepancy

The CLI applies the checkpoint's chat template. Rendering the same request through it and running the oracle gives `<think>\nOkay, the user is asking about the Fibonacci sequence beginning...`, which is the `"Okay, so the question is..."` output you saw. The CLI and the server were not disagreeing about the same input; the server had the raw prompt and matched a direct library run token for token.

## Chat-templated parity: token-exact

With the chat template the median top-2 gap rises from 0.625 to 3.875, bf16 and float32 oracles agree on all 32 tokens, and mlxcel is token-exact against both:

```
prompt ids  [128000, 128236, 837, 91949, 12082, 13328, 458, 128237]
mlxcel      [128227, 198, 37317, 11, 290, 1483, 371, 11935, 913, 290, 91949, 12082, 7963, 13, 6846, 611, 1311, 603, 125377, 1165, 290, 91949, 12082, 371, 13, 5902, 261, 4326, 324, 6578, 1551, 1981]
oracle      identical, in both bf16 and float32
```

`In distributed systems, consensus protocols such as Raft` through the chat template is also 32 of 32.

## The PR premise was wrong and the detection split is gone

Running mlxcel's existing DeepSeek-V2 decoder on the same checkpoint: chat-templated, 32 of 32 identical to the oracle; raw prompt, identical to the oracle's first 18 tokens, which is the full `" 1,1, and each subsequent number is the sum of the two preceding numbers."` The DeepSeek-V2 route handles this export correctly, so the `architectures[0]` split I added had nothing to fix and would only have moved working checkpoints onto a different decoder. It is removed, `deepseek_v2` keeps its route unconditionally, and `deepseek_v2_label_keeps_the_deepseek_v2_route` pins the Youtu relabel, a genuine `DeepseekV2ForCausalLM`, and a config with no `architectures` array, with the reasoning recorded so it is not re-added. That also restores the issue's original acceptance criterion, which said this checkpoint already loads through the existing DeepSeek-V2 route.

I should have measured that before encoding it in a routing rule rather than after; the report's section 5.5 records it.

## Tests added

`prefill_and_decode_paths_agree_position_by_position` pins the invariant the diagnosis turned on: the absorbed `l == 1` decode arm and the multi-token prefill arm must agree at every position on a synthetic model. Nothing else in the suite exercised the decode arm. `deepseek_v2_label_keeps_the_deepseek_v2_route` is the detection guard above.

The gate commands in the PR body now use the chat template and a config-rewritten copy of the local checkpoint, since the local directory declares `deepseek_v2` and therefore keeps the DeepSeek-V2 route by design.
